core/src/test/scala/kafka/utils/LoggingTest.scala

```diff
@@ -58,4 +60,20 @@ class LoggingTest extends Logging {
 
     assertEquals(logging.getClass.getName, logging.log.underlying.getName)
   }
+
+  @Test
+  def testLoggerLevelIsResolved(): Unit = {
+    val controller = new Log4jController()
+    val previousLevel = controller.getLogLevel("kafka")
+    try {
+      controller.setLogLevel("kafka", "TRACE")
```

Comment:
@tombentley A small propsal: would `org.apache.log4j.Level.TRACE.toString` be better?

It passes compilation checks.

![20200909-223825](https://user-images.githubusercontent.com/2375128/92605881-3fd02d00-f2ed-11ea-85d8-bf7a9e23cb90.png)

Reply:
@dongjinleekr that would be fine for the `setLogLevel()` and `assertEquals` calls, but there would still be the level name in the `assertTrue(...contains("kafka=TRACE"))` calls. I think fixing it there would make the code harder to read, and given that the declaration of `TRACE` is in an external API on which we depend, and the name will not change there doesn't seem to be any benefit to using the declaration rather than the name as a literal.

Reply:
@tombentley Great. Your stance is reasonable. I am convinced.